fvm/executionParameters.go

```diff
@@ -0,0 +1,137 @@
+package fvm
+
+import (
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+
+	"github.com/onflow/flow-go/fvm/blueprints"
+	"github.com/onflow/flow-go/fvm/errors"
+	"github.com/onflow/flow-go/fvm/meter/weighted"
+	"github.com/onflow/flow-go/fvm/utils"
+)
+
+// getExecutionWeights reads stored execution effort weights from the service account
+func getExecutionEffortWeights(
+	env Environment,
+	service runtime.Address,
+) (
+	computationWeights weighted.ExecutionEffortWeights,
+	err error,
+) {
+	value, err := env.VM().Runtime.ReadStored(
+		service,
+		cadence.Path{
+			Domain:     blueprints.TransactionExecutionParametersPathDomain,
+			Identifier: blueprints.TransactionFeesExecutionEffortWeightsPathIdentifier,
+		},
+		runtime.Context{Interface: env},
+	)
+	if err != nil {
+		// this might be fatal, return as is
+		return nil, err
+	}
+
+	computationWeightsRaw, ok := utils.CadenceValueToWeights(value)
+	if !ok {
+		// this is a non-fatal error. It is expected if the weights are not set up on the network yet.
+		return nil, errors.NewCouldNotGetExecutionParameterFromStateError(
+			service.Hex(),
+			blueprints.TransactionExecutionParametersPathDomain,
+			blueprints.TransactionFeesExecutionEffortWeightsPathIdentifier)
+	}
+
+	// Merge the default weights with the weights from the state.
+	// This allows for weights that are not set in the state, to be set by default.
+	// In case the network is stuck because of a transaction using an FVM feature that has 0 weight
+	// (or is not metered at all), the defaults can be changed and the network restarted
+	// instead of trying to change the weights with a transaction.
+	computationWeights = make(weighted.ExecutionEffortWeights)
```

Comment:
small optimisation - we can set a size here

Reply:
[389fb3c](https://github.com/onflow/flow-go/pull/2474/commits/389fb3cc0adbb7727024a2e00e337500c1994894)